app/controllers/user.py

```diff
@@ -110,6 +113,18 @@ async def index(display_name: Annotated[str, Path(min_length=1, max_length=DISPL
     groups_count = 0
     groups = ()
 
+    changesets_count_per_day = await ChangesetQuery.count_per_day_by_user_id(user.id, days=ACTIVITY_CHART_LENGTH)
+    dates_range = np.arange(
+        utcnow().replace(hour=0, minute=0, second=0, microsecond=0) - timedelta(days=ACTIVITY_CHART_LENGTH),
+        utcnow().replace(hour=0, minute=0, second=0, microsecond=0) + timedelta(days=1),
+        timedelta(days=1),
+    ).astype(datetime)
```

Comment:
Let's store time in a variable to ensure consistency, I have also found a 1-off error (ACTIVITY_CHART_LENGTH=365 returned 366 days).

```py
today = utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
created_since = today - timedelta(days=ACTIVITY_CHART_LENGTH - 1)
```

```suggestion
    dates_range = np.arange(
        created_since,
        today + timedelta(days=1),
        timedelta(days=1),
        dtype=datetime,
    )
```